src/router_core/connections.c

```diff
@@ -107,6 +130,61 @@ void qdr_connection_set_context(qdr_connection_t *conn, void *context)
         conn->user_context = context;
 }
 
+qdr_connection_info_t *qdr_connection_info(bool             is_encrypted,
+                                           bool             is_authenticated,
+                                           bool             opened,
+                                           char            *sasl_mechanisms,
+                                           const char      *dir,
```

Comment:
There is an enumerated type for direction.  You should use this rather than passing strings around.